processing/src/main/java/io/druid/segment/filter/AndFilter.java

```diff
@@ -80,10 +80,6 @@ public ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector)
   @Override
   public ValueMatcher makeMatcher(ColumnSelectorFactory factory)
   {
-    if (filters.size() == 0) {
-      return BooleanValueMatcher.of(false);
```

Comment:
This is odd because on one hand you end up with a value matcher that returned `false` here, but `makeMatcher` has a fall-through case of `true`.